js/containers/portal-dashboard/answer.tsx

```diff
@@ -18,34 +25,76 @@ class Answer extends React.PureComponent<IProps> {
-      { answer && (!question.get("required") || answer.get("submitted"))
-        ? this.renderAnswer()
-        : this.renderNoAnswer() }
-      </div>
-    );
-  }
-
-  private renderAnswer = () => {
-    const { answer, question } = this.props;
-    const type = answer.get("questionType");
-    const scored = question.get("scored");
-    const correct = scored ? answer.get("correct") : undefined;
+    const type = answer && answer.get("questionType");
+    const scored = question && question.get("scored");
+    const atype = answer && answer.get("type");
+    const qtype = question && question.get("type");
+    const correct = scored ? answer && answer.get("correct") : undefined;
     const answerType = AnswerTypes.find(at => at.type === type && at.correct === correct);
+    const questionType = QuestionTypes.find(qt => qt.type === qtype && qt.scored === scored);
     const searchRegExp = / /g;
     const iconId = answerType ? answerType.name.toLowerCase().replace(searchRegExp, "-") : "";
-    const AnswerIcon = answerType?.icon;
     return (
-      <div className={css.answerContent} data-cy={iconId}>
-        <AnswerIcon className={css.icon} />
+      <div className={css.answer} data-cy="student-answer">
+        {answer && (!question.get("required") || answer.get("submitted"))
+          ? this.renderAnswer(answerType?.icon, iconId, atype)
+          : this.renderNoAnswer(questionType?.icon, iconId)
+        }
       </div>
     );
   }
 
-  renderNoAnswer = () => {
+  private renderAnswer = (answerType: string, iconId: string, type: string) => {
+    const { inDetail } = this.props;
+    const AnswerIcon = answerType;
+
+    if (!inDetail) {
+      return (
+        <div className={css.answerContent} data-cy={iconId}>
+          <AnswerIcon />
+        </div>
+      );
+    } else {
+      return (
+        <div className={css.answerDetail}>
+          {this.renderAnswerDetail(type)}
+        </div>
+      );
+    }
+  }
+
+  renderNoAnswer = (questionType: string, iconId: string) => {
+    const { inDetail } = this.props;
+    const QuestionIcon = questionType;
+    if (!inDetail) {
+      return (
+        <div className={`${css.answerContent} ${css.noAnswer}`} data-cy={"no-answer"} />
+      );
+    } else {
+      return (
+        <div className={`${css.answerDetail} ${css.noAnswer}`} data-cy={iconId}>
+          <QuestionIcon />
+          No response
+        </div>
+      );
+    }
+  }
+
+  renderAnswerDetail = (type: string) => {
+    const { answer, question } = this.props;
+    const AnswerComponent: any = {
+      "multiple_choice_answer": MultipleChoiceAnswer,
+      "open_response_answer": OpenResponseAnswer,
+      "image_question_answer": ImageAnswer,
+      "external_link": ExternalLinkAnswer,
+      "interactive_state": IframeAnswer,
+    };
+    let AComponent;
+    if (answer && (!question.get("required") || answer.get("submitted"))) {
+      AComponent = AnswerComponent[type];
+    }
```

Comment:
I'd probably ternary this to shorten it.  Something like:
`const AComponent = (answer && (!question.get("required") || answer.get("submitted"))) ? AnswerComponent[type] : undefined;`

Not sure if it should be undefined or null.